File: year2/semestrul2/MPP/Laboratoare/AtletismConcurs_CSharp/AtletismConcurs_CSharp/Model/Participant.cs
namespace AtletismConcurs_CSharp.Model
{
    public class Participant
    {
        public string IdParticipant { get; set; }
        public string Nume { get; set; }
        public string Prenume { get; set; }
        public string Varsta { get; set; }
        public string Email { get; set; }

        public Participant(string idParticipant, string nume, string prenume, string varsta, string email)
        {
            IdParticipant = idParticipant;
            Nume = nume;
            Prenume = prenume;
            Varsta = varsta;
            Email = email;
        }

        public override bool Equals(object obj)
        {
            if (this == obj) return true;
            if (obj == null || GetType() != obj.GetType()) return false;
            Participant that = (Participant)obj;
            return IdParticipant.Equals(that.IdParticipant) &&
                   Nume.Equals(that.Nume) &&
                   Prenume.Equals(that.Prenume) &&
                   Varsta.Equals(that.Varsta) &&
                   Email.Equals(that.Email);
        }

        public override int GetHashCode()
        {
            return HashCode.Combine(IdParticipant, Nume, Prenume, Varsta, Email);
        }

        public override string ToString()
        {
            return $"Participant{{idParticipant='{IdParticipant}', nume='{Nume}', prenume='{Prenume}', varsta='{Varsta}', email='{Email}'}}";
        }
    }
}

File: year2/semestrul2/MPP/Laboratoare/lab4recuperare/src/main/java/org/example/Model/Participant.java
package org.example.Model;

import java.util.Objects;

public class Participant {
    private String id_participant;
    private String nume;
    private String prenume;
    private String varsta;
    private String email;

    public Participant(String id_participant, String nume, String prenume, String varsta, String email) {
        this.id_participant = id_participant;
        this.nume = nume;
        this.prenume = prenume;
        this.varsta = varsta;
        this.email = email;
    }

    public String getId_participant() {
        return id_participant;
    }

    public void setId_participant(String id_participant) {
        this.id_participant = id_participant;
    }

    public String getNume() {
        return nume;
    }

    public void setNume(String nume) {
        this.nume = nume;
    }

    public String getPrenume() {
        return prenume;
    }

    public void setPrenume(String prenume) {
        this.prenume = prenume;
    }

    public String getVarsta() {
        return varsta;
    }

    public void setVarsta(String varsta) {
        this.varsta = varsta;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Participant)) return false;
        Participant that = (Participant) o;
        return id_participant.equals(that.id_participant) &&
                nume.equals(that.nume) &&
                prenume.equals(that.prenume) &&
                varsta.equals(that.varsta) &&
                email.equals(that.email);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id_participant, nume, prenume, varsta, email);
    }

    @Override
    public String toString() {
        return "Participant{" +
                "id_participant='" + id_participant + '\'' +
                ", nume='" + nume + '\'' +
                ", prenume='" + prenume + '\'' +
                ", varsta='" + varsta + '\'' +
                ", email='" + email + '\'' +
                '}';
    }
}